File: Tests/EqualsTests.fs
namespace Extensions

open System
open Stringier
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type EqualsTests() =
    [<TestMethod>]
    member _.``mixed String/Span`` () =
        let mutable string = "hello"
        let mutable span = "hello".AsSpan()
        Assert.IsTrue(string.Equals(span, StringComparison.InvariantCulture))
        Assert.IsTrue(span.Equals(string, StringComparison.InvariantCulture))
        string <- "encyclopædia"
        span <- "encyclopaedia".AsSpan()
        Assert.IsTrue(string.Equals(span, StringComparison.InvariantCulture))
        Assert.IsTrue(span.Equals(string, StringComparison.InvariantCulture))

File: Tests/EqualsTests.cs
using System;
using Stringier;
using Defender;
using Xunit;

namespace Tests {
	public class EqualsTests : Trial {
		[Theory]
		[InlineData("", "", true)]
		[InlineData("", "hello", false)]
		[InlineData("hello", "", false)]
		[InlineData("hello", "hello", true)]
		public void Equals_Mixed(String source, String other, Boolean expected) {
			Claim.That(source.Equals(other.AsSpan(), StringComparison.CurrentCulture)).Equals(expected);
			Claim.That(source.AsSpan().Equals(other, StringComparison.CurrentCulture)).Equals(expected);
		}
	}
}
